src/pevm.rs

```diff
@@ -328,6 +262,54 @@ pub fn execute_revm_sequential<S: Storage>(
     Ok(results)
 }
 
+fn canonize_account(
+    results: &mut [PevmTxExecutionResult],
+    address: Address,
+    initial_account_basic: AccountBasic,
+    write_history: BTreeMap<TxIdx, MemoryEntry>,
+) {
+    let mut current_account = initial_account_basic;
+
+    for (tx_idx, memory_entry) in write_history {
+        match memory_entry {
+            MemoryEntry::Data(_, MemoryValue::Basic(info)) => {
+                current_account.balance = info.balance;
+                current_account.nonce = info.nonce;
+            }
+            MemoryEntry::Data(_, MemoryValue::LazyBalanceAddition(addition)) => {
+                current_account.balance += addition;
+            }
+            _ => unreachable!(),
+        }
+
+        let result = unsafe { results.get_unchecked_mut(tx_idx) };
+        match result.state.entry(address) {
+            std::collections::hash_map::Entry::Occupied(occupied) => {
+                let account = occupied.into_mut();
```

Comment:
`result.state.entry(address).or_default()` is much better if we always expect an output value. Like we can replace this whole `match` with:
```rust
        let account = result.state.entry(address).or_default();
        if current_account.is_empty() {
            *account = None;
        } else if let Some(account) = account {
            account.basic.balance = current_account.balance;
            account.basic.nonce = current_account.nonce;
        } else {
            *account = Some(EvmAccount {
                basic: current_account.clone(),
                storage: Default::default(),
            });
        }
```
Plus comments of course.

Reply:
This simplification makes a lot of sense. Thanks!